java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -2129,9 +2164,13 @@ public final Builder append(ColumnVector columnVector) {
       assert columnVector.type == type;
       assert columnVector.offHeap.hostData != null;
 
-      data.copyFromHostBuffer(currentIndex * type.sizeInBytes, columnVector.offHeap.hostData.data,
-          0L,
-          columnVector.getRowCount() * type.sizeInBytes);
+      if (type == DType.STRING_CATEGORY || type == DType.STRING) {
+        throw new IllegalStateException("Appending a string column vector client side is not currently supported");
```

Comment:
Nit: Seems like `UnsupportedOperationException` or `IllegalArgumentException` is more appropriate here.